src/pages/Poster.tsx

```diff
@@ -46,16 +51,22 @@ const Poster = () => {
   const countDownRef = useRef<any>(null)
   const history = useHistory()
 
-  useStorage()
-
-  const [isPosterExpired, setIsPosterExpired] = useState(false)
-  const [secondsRemaining, setSecondsRemaining] = useState(QUESTION_TIME)
-
   const { loading, error, data, refetch } = useQuery<RandomMovieData>(
     RANDOM_MOVIE,
   )
 
-  const { setTimeRemaining, setCurrentImdbId } = useStore()
+  const { timeRemaining, setTimeRemaining, setCurrentImdbId } = useStore()
+
+  const [isPosterExpired, setIsPosterExpired] = useState(false)
+
+  // setup an effect that updates the currentImdbId in the global store
+  // as soon as the random movie poster has been returned by the api
+  useEffect(() => {
```

Comment:
I added this hook such that we always have the latest imdb id in the store as soon as it changes (and not only on page transitions, as that could lead to "race conditions").